website/views.py

```diff
@@ -295,7 +295,7 @@ def resolve_guid(guid, suffix=None):
     if not resource:
         raise HTTPError(http_status.HTTP_404_NOT_FOUND)
 
-    if not guid == resource._id:
+    if issubclass(resource.__class__, VersionedGuidMixin) and guid != resource._id:
```

Comment:
Good catch!

Note for myself: it should be equivalent to `if version and guid != resource._id:` but this needs `resource, version = Guid.load_referent(guid)`